### ISSUE TITLE
BinaryExpression breaking

### ISSUE THREAD

**Issue by belav** [belav/csharpier#92]
We may want to avoid breaking if there is a single binary expression?
```c#
                IsOneToMany = entityDefinition.Assignments.SingleOrDefault(
                    o => o.PropertyName == assignmentProperty
                )?.IsOneToMany
                ?? false, // should be on the line above probably
```

**Comment by belav** [comment]
See also this, which is probably long enough to break, but should probably break on the = first
```c#
            assignment.InstructionalText = editAssignmentViewModel.InstructionalText
            ?? string.Empty;
```

**Comment by belav** [comment]
See also
```c#
                var fixDiagnosticsMS =
                    analysisStopwatch.ElapsedMilliseconds
                    - projectDiagnosticsMS;
```

**Comment by belav** [comment]
See also
```c#
                if (
                    latestVersion == null
                    || (potentialLatestVersion != null
                    && potentialLatestVersion.CompareTo(latestVersion) > 0)
                ) {
```

**Comment by belav** [comment]
See also
```c#
                if (
                    members[i]
                        is IMethodSymbol method
                    && (IsConfigureServices(symbols, method) || IsConfigure(symbols, method))
                ) {
```

**Comment by belav** [comment]
See also - the ( ) do weird things
```c#
            return (optionsProvider != null
                && optionsProvider.TryGetDiagnosticValue(
                    tree,
                    diagnosticId,
                    CancellationToken.None,
                    out _
                ))
                || (diagnosticCategory != null
                && analyzerConfigOptions.TryGetValue(
                    GetCategoryBasedDotnetAnalyzerDiagnosticSeverityKey(diagnosticCategory),
                    out _
                ))
                || analyzerConfigOptions.TryGetValue(DotnetAnalyzerDiagnosticSeverityKey, out _);
```

**Comment by belav** [comment]
This should probably be combined with https://github.com/belav/csharpier/issues/37

**Comment by belav** [comment]
This was all addressed in #37
